prow/gerrit/adapter/adapter.go

```diff
@@ -278,15 +278,23 @@ func (c *Controller) ProcessChange(instance string, change client.ChangeInfo) er
 	case client.New:
 		presubmits := c.config().Presubmits[cloneURI.String()]
 		presubmits = append(presubmits, c.config().Presubmits[cloneURI.Host+"/"+cloneURI.Path]...)
-		for _, presubmit := range presubmits {
-			if shouldRun, err := presubmit.ShouldRun(change.Branch, changedFiles, false, false); err != nil {
-				return fmt.Errorf("failed to determine if presubmit %q should run: %v", presubmit.Name, err)
-			} else if shouldRun {
-				jobSpecs = append(jobSpecs, jobSpec{
-					spec:   pjutil.PresubmitSpec(presubmit, refs),
-					labels: presubmit.Labels,
-				})
-			}
+		var filters []pjutil.Filter
+		filter, err := messageFilter(c.lastUpdate, change, presubmits)
+		if err != nil {
+			logger.WithError(err).Warn("failed to create filter on messages for presubmits")
+		} else {
+			filters = append(filters, filter)
+		}
+		filters = append(filters, oldRevisionFilter(c.lastUpdate, change.Revisions[change.CurrentRevision]))
+		toTrigger, _, err := pjutil.FilterPresubmits(pjutil.AggregateFilter(filters), listChangedFiles(change), change.Branch, presubmits, logger)
```

Comment:
Do we post any sort of `skipped` contexts today in Gerrit?

Reply:
Nope, Gerrit doesn't have a concept of contexts 😕

Reply:
yeah... gerrit doesn't have context, gerrit doesn't allow edit comments, :disappointed: we can improve things when gerrit adds more features (in the future)...